packages/desktop/src/services/FileDownloadServiceElectron.ts

```diff
@@ -352,4 +534,32 @@ export default class FileDownloadServiceElectron
             });
         });
     }
+
+    private parseS3Url(url: string): { bucket: string; key: string; region: string } {
+        const { hostname, pathname } = new URL(url);
+        const [bucket] = hostname.split(".");
+        const key = pathname.slice(1);
+        const region = hostname.split(".")[2];
+        return { bucket, key, region };
+    }
+
+    private async listS3Objects(bucket: string, prefix: string, region: string): Promise<string[]> {
+        const url = `https://${bucket}.s3.${region}.amazonaws.com?list-type=2&prefix=${encodeURIComponent(
+            prefix
+        )}`;
+        const response = await axios.get(url);
+        const parsedResult = await parseStringPromise(response.data);
+
+        const contents = parsedResult.ListBucketResult.Contents || [];
+        const keys: string[] = [];
+
+        for (const content of contents) {
```

Comment:
FWIW Could be more ~ javascripty ~ using more innate array toolings
```
return contents
    .map(content => content.Key?.[0])
    .filter(key => key === "string");
```

Reply:
resolved in [cfdf9b4](https://github.com/AllenInstitute/biofile-finder/pull/204/commits/cfdf9b40628c86077ac4ad73f158267bde3b5b49) by combining logic. needs more testing though once the caching issue is fixed.